src/cli/command-processor.ts

```diff
@@ -0,0 +1,210 @@
+import {Worker} from 'worker_threads';
+import yargs from 'yargs/yargs';
+import * as readline from 'readline';
+import {CommandMessage, READY_FOR_MESSAGE, CHECK_READY} from './commands';
+import {ExecuteMessage, IdentifyMessage, ScanMessage} from './commands';
+import {UDPScanConfig} from '../radio/radio-controller';
+
+/**
+ * Class to parse raw user input from stdin.
+ */
+export class CommandProcessor {
+  private worker: Worker;
+  constructor(worker: Worker) {
+    this.worker = worker;
+  }
+
+  /**
+   * Parses a string and returns a CommandMessage, if valid.
+   * @param userCommand  A string to parse.
+   * @returns  A promise that resolves to a valid `CommandMessage`,
+   *   or void if an `exit` command was detected
+   */
+  private async parseIntent(
+    userCommand: string
+  ): Promise<CommandMessage | void> {
+    const argv = await yargs()
+      .command(
+        'udp-scan',
+        'Scan and identify devices with a given UDP scan configuration.',
+        yargs => {
+          return yargs
+            .option('broadcast_address', {
+              describe: 'Destination IP address foir the UDP broadcast.',
+              type: 'string',
+              demandOption: true,
+            })
+            .option('broadcast_port', {
+              describe: 'The destination port for the UDP discovery broadcast.',
+              type: 'number',
+              demandOption: true,
+            })
+            .option('listen_port', {
+              describe: 'The port to listen for the UDP discvery response.',
+              type: 'number',
+              demandOption: true,
+            })
+            .option('discovery_packet', {
+              describe: 'The payload to send in the UDP broadcast.',
+              type: 'string',
+              demandOption: true,
+            })
+            .option('request_id', {
+              describe: 'An optional request ID for the Identify Request.',
+              type: 'string',
+              default: 'default-request-id',
+              demandOption: false,
+            })
+            .option('device_id', {
+              describe: 'The device ID to include in the Identify Request.',
+              type: 'string',
+              default: 'default-device-id',
+              demandOption: false,
+            });
+        }
+      )
+      .command(
+        'simulate-identify',
+        'Manually trigger an Identify response.',
+        yargs => {
+          return yargs
+            .option('request_id', {
+              describe: 'The request Id',
+              type: 'string',
+              demandOption: true,
+            })
+            .option('discovery_buffer', {
+              describe: 'The IDENTIFY dicovery buffer represented as a string',
```

Comment:
discovery

Reply:
Done.  Will see if there is some global spellcheck I can run for comments.